src/wp-admin/includes/class-wp-filesystem-direct.php

```diff
@@ -331,12 +338,18 @@ public function move( $source, $destination, $overwrite = false ) {
 			return false;
 		}
 
+		if ( $overwrite && $this->exists( $destination ) && ! $this->delete( $destination, true ) ) {
+			// Can't overwrite if the destination couldn't be deleted.
+			return false;
+		}
+
 		// Try using rename first. if that fails (for example, source is read only) try copy.
 		if ( @rename( $source, $destination ) ) {
 			return true;
 		}
 
-		if ( $this->copy( $source, $destination, $overwrite ) && $this->exists( $destination ) ) {
+		// Backwards Compatibility: Only fall back to `::copy()` for single files.
```

Comment:
```suggestion
		// Backward compatibility: Only fall back to `::copy()` for single files.
```
Per [spelling recommendations](https://make.wordpress.org/core/handbook/best-practices/spelling/).